examples/server/server.cpp

```diff
@@ -2366,7 +2518,11 @@ struct server_context {
                         // release slot because of stop condition
                         slot.release();
                         slot.print_timings();
+                        #ifndef OAI_FULL_COMPAT
                         send_final_response(slot);
+                        #else
+                        send_final_response_oaicompat(slot);
```

Comment:
the `send_final_*` is called from inference thread, but what we're doing is only to format the response. I'd suggest to move your code to a new function `format_final_*_oaicompat`, much like what we have with `format_final_response_oaicompat`

Reply:
I noticed this yesterday. I found that there's a function called handle_completions_generic (there was a TODO suggesting merging that with handle_chat_completions. I've done that. I'll create another PR with that since it's probably the right way to do it.

I can probably do the `oai_compat`set to true by default later and send a PR draft